### ISSUE TITLE
permissions for calling AWS SNS APIs within a Zappa app

### ISSUE THREAD

**Issue by dtenenba** [Miserlou/Zappa#181]
I have a Zappa app that uses boto3 to publish a message to an SNS topic. 
This causes an error. Here's an excerpt from `zappa tail dev`, with some info elided:

```
ClientError: An error occurred (AuthorizationError) when calling the Publish operation: 
User: arn:aws:sts::XXX:assumed-role/ZappaLambdaExecution/awslambda_XXX is not authorized
to perform: SNS:Publish on resource: arn:aws:sns:us-west-2:XXX:lambda-trigger-notifier
```

Does Zappa look at my code to see which APIs I am calling and automatically configure the IAM user to have permission to call those APIs? If so, it does not seem to be working. If not, what is the recommended way to give the IAM role permission to call that API on that resource?

**Comment by dtenenba** [comment]
For the record, I fixed this by modifying the policy of the IAM role `zappa-permissions` to have permission to perform the action `SNS:Publish` on my SNS topic.

**Comment by dtenenba** [comment]
Oops, but to amend the record, this change to the IAM role disappeared the next time I did `zappa update dev`. How can I persist IAM settings?

**Comment by Miserlou** [comment]
Oh wow, that's two issues right there!

Reading and writing to SNS should be part of the IAM permissions of Zappa IAM. Can you post the syntax needed for that? sns::*, I suppose?

The second is that there needs to be a configurable option to NOT override the IAM role if one is explicitly specified. 

Thanks for bringing this up!

**Comment by dtenenba** [comment]
Solved this, sorry for the noise. I created a new policy based on the policy in the `zappa-permissions` policy, then added the extra permissions I needed. Then I created a new role based on that policy and specified that role as `role_name` in my `zappa_settings.json`. I had to undeploy and deploy the app but then it worked.

So I think things are ok but will leave this open if you think there is still one or more issues.
